Controllers/ProductController.cs

```diff
@@ -39,5 +38,24 @@ public async Task<IActionResult> AddAsync([FromBody] ProductDTO product)
                     return StatusCode(StatusCodes.Status500InternalServerError, ex.Message);
               }
         }
+
+        [HttpGet]
+        public async Task<IActionResult> GetProductsAsync()
+        {
+            try
+            {
+                var products = await _productService.GetProductsAsync();
+                if (products != null)
```

Comment:
return Ok(await _productService.GetProductsAsync()); is sufficient.

Reply:
no need to check any null condition?

Reply:
I believe that statement gives you NoContent Result, When the result is null, Could you please check?